src/main/java/com/artipie/api/RepositoryRest.java

```diff
@@ -205,4 +205,34 @@ private void removeRepo(final RoutingContext context) {
                 .end();
         }
     }
+
+    /**
+     * Move a repository settings.
+     * @param context Routing context
+     */
+    private void moveRepo(final RoutingContext context) {
+        final RepositoryName rname = new RepositoryName.FromRequest(context, this.layout);
+        final ReservedNamesVerifier reserved = new ReservedNamesVerifier(rname);
+        final SettingsDuplicatesVerifier duplicates =
+            new SettingsDuplicatesVerifier(rname, this.crs);
+        final ExistenceVerifier existence = new ExistenceVerifier(rname, this.crs);
+        final Validator validator = validator(
+            validator(reserved::valid, reserved::message, HttpStatus.BAD_REQUEST_400),
+            validator(existence::valid, existence::message, HttpStatus.NOT_FOUND_404),
+            validator(duplicates::valid, duplicates::message, HttpStatus.CONFLICT_409)
+        );
+        if (validator.validate(context)) {
+            final String newname = BaseRest.readJsonObject(context).getString("name");
+            final RepositoryName newrname;
+            if (new Layout.Flat().toString().equals(this.layout)) {
+                newrname = new RepositoryName.Flat(newname);
+            } else {
+                newrname = new RepositoryName.Org(newname, context.pathParam(RepositoryName.UNAME));
+            }
```

Comment:
it seems, that new repository name should be validated the way as the old one...

Reply:
@olenagerasimova thanks, I have added changes, please review